src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -50,6 +50,55 @@ struct LQPOutputRelation {
 
 using LQPNodeMapping = std::unordered_map<std::shared_ptr<const AbstractLQPNode>, std::shared_ptr<AbstractLQPNode>>;
 
+/**
+ * Models a functional dependency (FD), which consists out of two sets of column expressions (Left &
+ * Right).
+ * The left column set unambigiously identifies the right column set: {Left} => {Right}
+ *
+ * Example A:
+ * Think of a table with three columns: (Semester, CourseID, Lecturer)
+ * The primary key is defined across the first two columns, which leads to the following FD:
+ * {Semester, CourseID} => {Lecturer}
+ *
+ * Example B:
+ * Think of a table with four columns: (ISBN, Genre, Author, Author-Nationality)
+ * The primary key {ISBN} identifies all other columns. Therefore, we get the following FDs:
+ * {ISBN} => {Author}
+ * {ISBN} => {Genre}
+ * {ISBN} => {Author}
+ * {ISBN} => {Author-Nationality}
+ * Furthermore, knowing an author, we also know his nationality. Hence, we can specify another FD that applies:
+ * {Author} => {Author-Nationality}
+ *
+ *
+ * Currently, column expressions are required to be non-nullable to be involved in FDs. As there are
+ * strategies to combine both, null values and FDs (e.g. https://arxiv.org/abs/1404.4963), this might
+ * change in the future.
+ */
+struct FunctionalDependency : std::pair<ExpressionUnorderedSet, ExpressionUnorderedSet> {
```

Comment:
Inheriting from STL containers is very dangerous, especially if it is public inheritance.

Reply:
Noted, thanks!